File: backend/app/schemas/life.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, field_serializer

from app.core.time import to_app_timezone


class LifeRecordItem(BaseModel):
    id: int
    session_id: str
    title: Optional[str] = None
    content: str
    mood_label: Optional[str] = None
    location: Optional[str] = None
    tags: list[str] = []
    media_url: Optional[str] = None
    created_at: datetime

    @field_serializer("created_at")
    def serialize_created_at(self, value: datetime) -> str:
        return to_app_timezone(value).isoformat()

File: frontend/src/api/life.ts
import axios from 'axios'

const api = axios.create({ baseURL: '/api', timeout: 60000 })

export interface LifeRecordItem {
  id: number
  session_id: string
  title?: string | null
  content: string
  mood_label?: string | null
  location?: string | null
  tags: string[]
  media_url?: string | null
  created_at: string
}

export const createLifeRecord = (form: FormData) =>
  api.post<LifeRecordItem>('/life/records', form)

export const getLifeRecords = (sid: string) =>
  api.get<LifeRecordItem[]>(`/life/records?session_id=${sid}`)

export const deleteLifeRecord = (id: number, sid: string) =>
  api.delete(`/life/records/${id}?session_id=${sid}`)
